v3/setup/etcdctl-auth.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+source /etc/environment
+
+etcdctl set /etcdctl/config/root-user root
+etcdctl set /etcdctl/config/read-user read-user
+etcdctl set /etcdctl/config/write-user write-user
+
+ROOT_USERNAME=$(etcdctl get /etcdctl/config/root-user)
+ROOT_PASSWORD=$(etcdctl get /etcdctl/config/root-password)
+READ_USERNAME=$(etcdctl get /etcdctl/config/read-user)
+READ_PASSWORD=$(etcdctl get /etcdctl/config/read-password)
+WRITE_USERNAME=$(etcdctl get /etcdctl/config/write-user)
+WRITE_PASSWORD=$(etcdctl get /etcdctl/config/write-password)
+
+CRED_DIR="/opt/etcdctl"
+if [[ ! -d $CRED_DIR ]]; then
+    sudo mkdir $CRED_DIR -p
+fi
+
+# add root-user
+sudo echo '{"user":"'${ROOT_USERNAME}'", "password":"'${ROOT_PASSWORD}'"}' > $CRED_DIR/root-user.json
+curl -L http://127.0.0.1:2379/v2/auth/users/${ROOT_USERNAME} -XPUT -d "@$CRED_DIR/root-user.json"
+
+# add read-user
+sudo echo '{"user":"'${READ_USERNAME}'", "password":"'${READ_PASSWORD}'"}' > $CRED_DIR/read-user.json
```

Comment:
@anshupande I think we're better off using a heredoc instead of using echo like this as this could end up being error prone (you missing a quote etc)

so the line above can be rewritten as something like

```
sudo cat << EOF > $CRED_DIR/read-user.json
{
  "user": "${READ_USERNAME}",
  "password": "${READ_PASSWORD}"
}
EOF
```
It's also a lot easier to read/troubleshoot